mmengine/config/new_config.py

```
@@ -0,0 +1,459 @@
+                self.lazy_importers.append(p)
+                p.enable = False
+
+        old_import = builtins.__import__
+
+        def new_import(name, globals=None, locals=None, fromlist=(), level=0):
+            cur_file = None
+
+            # Try to import the base config source file
+            if level != 0 and globals is not None:
+                # For relative import path
+                if '__file__' in globals:
+                    loc = Path(globals['__file__']).parent
+                else:
+                    loc = Path(os.getcwd())
+                cur_file = self.find_relative_file(loc, name, level - 1)
+                if not cur_file.exists():
+                    raise ImportError(
+                        f'Cannot find the base config "{name}" from '
+                        f'{loc}: {cur_file} does not exist.')
+            elif level == 0:
+                # For absolute import path
+                pkg, _, mod = name.partition('.')
+                pkg = PathFinder.find_spec(pkg)
+                if mod and pkg.submodule_search_locations:
+                    loc = Path(pkg.submodule_search_locations[0])
+                    cur_file = self.find_relative_file(loc, mod)
+                    if not cur_file.exists():
+                        raise ImportError(
+                            f'Cannot find the base config "{name}": '
+                            f'{cur_file} does not exist.')
+
+            # Recover the original import during handle the base config file.
+            builtins.__import__ = old_import
+
+            if cur_file is not None:
+                mod = ConfigV2._get_config_module(cur_file)
+
+                for k in dir(mod):
+                    mod.__dict__[k] = ConfigV2._dict_to_config_dict_lazy(
+                        getattr(mod, k))
+            else:
+                mod = old_import(
+                    name, globals, locals, fromlist=fromlist, level=level)
```

syntax like `import torch` should be illegal in `with read_base`. Maybe raise an error here could be better

---

Why it's illegal?